graylog2-server/src/main/java/org/graylog/events/processor/aggregation/AggregationEventProcessor.java

```diff
@@ -235,10 +239,14 @@ private void filterSearch(EventFactory eventFactory, AggregationEventProcessorPa
                         .streams(event.getSourceStreams())
                         .build());
 
+                LOG.debug("Creating event {}/{} {}", eventDefinition.title(), eventDefinition.id(), msg.getTimestamp());
                 eventsWithContext.add(EventWithContext.create(event, msg));
             }
 
-            eventsConsumer.accept(eventsWithContext.build());
+            List<EventWithContext> eventsList = eventsWithContext.build();
+            eventDefinitionHandler.updateLastMatched(eventsList);
```

Comment:
Couldn't this maybe be moved into `emitEvents` of `EventProcessorEngine`?

This would remove the duplicate code in the different processors.

Reply:
Indeed. I expected event processors would require some specific code based on event type, but that wasn't the case. Much simpler this way.